rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/90_sparse_vector.yml

```diff
@@ -163,16 +163,6 @@
 
   - match: { acknowledged: true }
 
-  - do:
-      catch: /\[sparse_vector\] fields do not support \[exists\] queries/
-      search:
-        rest_total_hits_as_int: true
-        index: test
-        body:
-          query:
-            exists:
-              field: ml.tokens
```

Comment:
we shouldn't remove this. Behavior for 8.0.0-8.11.0 should not be changed.

Additionally, the test says ""Sparse vector in 7.x": and now you are restricting it to never be 7.x. That seems weird.

Please adjust the test so that it only tests 7.x & add one for [8.0.0-8.11.0) that includes the exists clause failure

Reply:
This was more of a retraction commit tbh, removing a test from the 7.x section introduced in the [original PR](https://github.com/elastic/elasticsearch/commit/e7fa8e8d1da834f77964ba3e6752dd4d3e2097be#diff-c78fb74190e4dc46908a039540fb664f6ead6ada1c6c639d498791c0ce742d85L103)  for exists queries in sparse vectors for 8.11. 

I was trying to limit the effect of the mixed node cluster setup just in the tests and try to account for the different scenarios under separate requests, experimenting also with the `node_selector` - but couldn't find any similar cases w/o any modifications to the codebase as well. I also tried limiting the nodes & setting `num_replicas=0` when creating the index for the 7.x scenario, however this would pretty much defeat the purpose of any cluster/bcw test altogether.  The issue with the node selector, AFAICT, was that when running the test we might end up with a replica placed on a recently upgraded 8.11 index, which later caused all searches that landed on that node to fail, as the index version was less than the expected [one for the mixed cluster](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java#L114), raising the exception, even when we had explicitly defined the `node_selector` for the search request.   

Ended up slightly  updating the codebase as well, to return a deprecation warning along with the headers, and treat incoming 7.x `exists` queries for `sparse_vector` fields as unmatchable ones on >= 8.11  (i.e. `return a MatchNoDocsQuery` instead). Happy to remove the deprecation & fallback to raising `IllegalArgumentException` in all cases, but would need some help in adjusting it with the existing restapi-test params, as I'm not familiar with the setup & alternatively, I had to do some refactoring on the tests to make it pass consistently. 🤔 

I also added an explicit case for [8.0, 8.11) versions though - expecting such requests to fail in all cases - which **should** be fine as the behaviour will be consistent for the **mixed-cluster** test suite.